modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,566 @@
+        ]
+
+    def apply(
+        self,
+        func,
+        *args,
+        num_splits=None,
+        other_axis_partition=None,
+        maintain_partitioning=True,
+        lengths=None,
+        manual_partition=False,
+        **kwargs,
+    ):
+        """
+        Apply a function to this axis partition along full axis.
+
+        Parameters
+        ----------
+        func : callable
+            The function to apply.
+        *args : iterable
+            Additional positional arguments to be passed in `func`.
+        num_splits : int, default: None
+            The number of times to split the result object.
+        other_axis_partition : PandasDataframeAxisPartition, default: None
+            Another `PandasDataframeAxisPartition` object to be applied
+            to func. This is for operations that are between two data sets.
+        maintain_partitioning : bool, default: True
+            Whether to keep the partitioning in the same
+            orientation as it was previously or not. This is important because we may be
+            operating on an individual AxisPartition and not touching the rest.
+            In this case, we have to return the partitioning to its previous
+            orientation (the lengths will remain the same). This is ignored between
+            two axis partitions.
+        lengths : list, optional
+            The list of lengths to shuffle the object.
+        manual_partition : bool, default: False
+            If True, partition the result with `lengths`.
+        **kwargs : dict
+            Additional keywords arguments to be passed in `func`.
+
+        Returns
+        -------
+        list
+            A list of `PandasOnUnidistDataframeVirtualPartition` objects.
+
+        Notes
+        -----
+        In older versions of Modin, ``args`` was passed internally as ``kwargs["args"]``, and
+        deserialization was handled in a special case in ``deploy_unidist_func``, making control flow
+        difficult to follow. All deserialization is still handled in ``deploy_unidist_func``, but
+        in a more direct fashion.
+        """
+        if not self.full_axis:
+            # If this is not a full axis partition, it already contains a subset of
+            # the full axis, so we shouldn't split the result further.
+            num_splits = 1
+        if len(self.call_queue) > 0:
+            self.drain_call_queue()
+        result = super(PandasOnUnidistDataframeVirtualPartition, self).apply(
+            func,
+            *args,
+            num_splits=num_splits,
+            other_axis_partition=other_axis_partition,
+            maintain_partitioning=maintain_partitioning,
+            lengths=lengths,
+            manual_partition=manual_partition,
+            **kwargs,
+        )
+        if self.full_axis:
+            return result
+        else:
+            # If this is a full axis partition, just take out the single split in the result.
+            return result[0]
```

Comment:
comment seems to be contradicting the condition

Reply:
The comment is aligned with the comment above. I think it is okay.

Reply:
fixed